connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Herder.java

```
@@ -82,7 +82,7 @@
     void connectorConfig(String connName, Callback<Map<String, String>> callback);
 
     /**
-     * Set the configuration for a connector. This supports creation, update, and deletion.
+     * Set the configuration for a connector. This supports creation and updating.
```

Given this change, should we be checking for null configs in `ConnectorsResource.putConnectorConfig()` and redirecting to `removeConnector` if needed?

---

Or perhaps we should raise an exception if the config is null? Are you worried about people who have coded against the `Herder` directly?

---

No, I was worried that previously we handled the `null` as a deletion, so `ConnectorsResource.putConnectorConfig()` only calling `Herder.putConnectorConfig()` may have changed behavior -- previously with DistributedHerder the deletion would have worked whereas now it will throw an exception.

---

Makes sense. So do we want to continue to allow `ConnectorsResource.putConnectorConfig()` to support deletion? I think I'd probably prefer to require the user call DELETE explicitly, but I'm not sure if anyone depends on this behavior at the moment.

---

Looking at this a little more, `ConnectorsResource.putConnectorConfig()` would throw NPE if the config is null, so I don't think it could have been used for deletion. So maybe we're OK?

---

Yeah, if we're confident it's not a compatibility breaking thing, then I agree the explicit `DELETE` is better.